src/compiler/config/outputs/validate-hydrate-script.ts

```diff
@@ -59,7 +58,9 @@ export const validateHydrateScript = (config: d.ValidatedConfig, userOutputs: d.
 
     outputTarget.external = outputTarget.external || [];
 
-    outputTarget.external.push(...NODE_BUILTINS);
+    outputTarget.external.push('fs');
+    outputTarget.external.push('path');
+    outputTarget.external.push('crypto');
```

Comment:
or wait got confused - yeah I got errors, but I think we need to leave it as-is because the `NODE_BUILTINS` is now found in the `script/` dir, not in the `src/` tree

Reply:
Do the karma tests give us sufficient coverage for that, or do we need to do something in addition to them?

Reply:
We don't unfortunately. We have a few things that verify it builds and outputs files of a specific name, but nothing with full on karma tests.

- https://github.com/ionic-team/stencil/tree/main/test/prerender-shadow doesn't have any tests, but should be built during CI (so in theory you could start it locally and poke at it)
- https://github.com/ionic-team/stencil/blob/main/test/end-to-end/test-end-to-end-hydrate.js is probably the best programmatic tests rn

Reply:
FWIW I smoke tested `dist-hydrate-script` (which when run, outputs to a `hydrate/` dir with the following express server script:
```js
const hydrate = require('../hydrate');

const express = require('express')
const app = express()
const port = 3000

app.get('/', (req, res) => {
  const result = hydrate.renderToString(`<div>hello world<my-component first="SSR"></my-component></div>`).then((r) => { console.log(r.html); res.send(r.html) })
  
})

app.listen(port, () => {
  console.log(`Example app listening on port ${port}`)
})
```

Reply:
I'm assuming it more or less worked then? haha